src/prop/prop_engine.cpp

```diff
@@ -593,10 +618,30 @@ std::shared_ptr<ProofNode> PropEngine::getProof()
   {
     return nullptr;
   }
+  if (options::unsatCores()
```

Comment:
I'm not sure if this really is the right place to return this since we can't distinguish between returning a proof step for the unsat core or the proof itself.